Modules/EMCAL/src/CellOccupancyCheck.cxx

```diff
@@ -0,0 +1,251 @@
+          SMcounts[3] += counts;
+        }
+        if (BinEdgesX[8] <= bin_low_x && bin_up_x <= BinEdgesX[9] && BinEdgesY[8] <= bin_low_y && bin_up_y <= BinEdgesY[9]) {
+          SMcounts[4] += counts;
+        }
+        if (BinEdgesX[10] <= bin_low_x && bin_up_x <= BinEdgesX[11] && BinEdgesY[10] <= bin_low_y && bin_up_y <= BinEdgesY[11]) {
+          SMcounts[5] += counts;
+        }
+        if (BinEdgesX[12] <= bin_low_x && bin_up_x <= BinEdgesX[13] && BinEdgesY[12] <= bin_low_y && bin_up_y <= BinEdgesY[13]) {
+          SMcounts[6] += counts;
+        }
+        if (BinEdgesX[14] <= bin_low_x && bin_up_x <= BinEdgesX[15] && BinEdgesY[14] <= bin_low_y && bin_up_y <= BinEdgesY[15]) {
+          SMcounts[7] += counts;
+        }
+        if (BinEdgesX[16] <= bin_low_x && bin_up_x <= BinEdgesX[17] && BinEdgesY[16] <= bin_low_y && bin_up_y <= BinEdgesY[17]) {
+          SMcounts[8] += counts;
+        }
+        if (BinEdgesX[18] <= bin_low_x && bin_up_x <= BinEdgesX[19] && BinEdgesY[18] <= bin_low_y && bin_up_y <= BinEdgesY[19]) {
+          SMcounts[9] += counts;
+        }
+        if (BinEdgesX[20] <= bin_low_x && bin_up_x <= BinEdgesX[21] && BinEdgesY[20] <= bin_low_y && bin_up_y <= BinEdgesY[21]) {
+          SMcounts[10] += counts;
+        }
+        if (BinEdgesX[22] <= bin_low_x && bin_up_x <= BinEdgesX[23] && BinEdgesY[22] <= bin_low_y && bin_up_y <= BinEdgesY[23]) {
+          SMcounts[11] += counts;
+        }
+        if (BinEdgesX[24] <= bin_low_x && bin_up_x <= BinEdgesX[25] && BinEdgesY[24] <= bin_low_y && bin_up_y <= BinEdgesY[25]) {
+          SMcounts[12] += counts;
+        }
+        if (BinEdgesX[26] <= bin_low_x && bin_up_x <= BinEdgesX[27] && BinEdgesY[26] <= bin_low_y && bin_up_y <= BinEdgesY[27]) {
+          SMcounts[13] += counts;
+        }
+        if (BinEdgesX[28] <= bin_low_x && bin_up_x <= BinEdgesX[29] && BinEdgesY[28] <= bin_low_y && bin_up_y <= BinEdgesY[29]) {
+          SMcounts[14] += counts;
+        }
+        if (BinEdgesX[30] <= bin_low_x && bin_up_x <= BinEdgesX[31] && BinEdgesY[30] <= bin_low_y && bin_up_y <= BinEdgesY[31]) {
+          SMcounts[15] += counts;
+        }
+        if (BinEdgesX[32] <= bin_low_x && bin_up_x <= BinEdgesX[33] && BinEdgesY[32] <= bin_low_y && bin_up_y <= BinEdgesY[33]) {
+          SMcounts[16] += counts;
+        }
+        if (BinEdgesX[34] <= bin_low_x && bin_up_x <= BinEdgesX[35] && BinEdgesY[34] <= bin_low_y && bin_up_y <= BinEdgesY[35]) {
+          SMcounts[17] += counts;
+        }
+        if (BinEdgesX[36] <= bin_low_x && bin_up_x <= BinEdgesX[37] && BinEdgesY[36] <= bin_low_y && bin_up_y <= BinEdgesY[37]) {
+          SMcounts[18] += counts;
+        }
+        if (BinEdgesX[38] <= bin_low_x && bin_up_x <= BinEdgesX[39] && BinEdgesY[38] <= bin_low_y && bin_up_y <= BinEdgesY[39]) {
+          SMcounts[19] += counts;
+        }
+      }
+    }
```

Comment:
This is a lot of checking. I wonder whether it wouldn't make more sense to iterate in the outer loop over the supermodule, and in the two inner ones about column and row, where with column and row you access the bin content.